src/components/element-editor/validation/validation-child.tsx

```diff
@@ -0,0 +1,84 @@
+import { Form } from "@lifesg/react-design-system";
+import { useState } from "react";
+import { ChildEntry } from "src/components/common";
+import { FieldWrapper } from "./validation.styles";
+import { IEntryProps } from "./validation";
+
+interface IProps {
+    onDelete: () => void;
+    options: string[];
+    onChange: (newValue: any) => void;
+}
+
+export const ValidationChild = ({ onDelete, options, onChange }: IProps) => {
+    // =========================================================================
+    // CONST, STATES, REFS
+    // =========================================================================
+    const [type, setType] = useState<string>(
+        options.length === 1 ? options[0] : ""
+    );
+
+    // =============================================================================
+    // EVENT HANDLERS
+    // =============================================================================
+    const handleChange = (rule?: string, errorMessage?: string) => {
+        const newValue = {
+            type: type,
+            rule: rule,
+            errorMessage: errorMessage,
+        };
+
+        onChange((prevValue: IEntryProps[]) => {
+            const prevValueWithSameType = prevValue.find(
+                (value: IEntryProps) => value.type === type
+            );
+            if (prevValueWithSameType) {
+                return prevValue.map((value: IEntryProps) =>
+                    value.type === type ? { ...value, ...newValue } : value
+                );
+            } else {
+                return [...prevValue, newValue];
+            }
+        });
+    };
```

Comment:
we want to maintain a separation of concern between the parent and child component. this means that we should not pass a parent's state setter to the child.

the child should only need to pass the rule and errorMessage to the parent. it will be up to the parent to manage the update to its own internal state.